pallets/rmrk-equip/src/functions.rs

```diff
@@ -302,11 +258,109 @@ where
 						Ok(())
 					},
 				)?;
-				Ok((item_collection_id, item_nft_id, base_id, slot_id, true))
+				Ok((item_collection_id, item_nft_id, base_id, slot_id))
 			},
 		}
 	}
 
+	/// Implementation of the do_unequip function for the Base trait
+	/// Called by the equip extrinsic to unequip a child NFT's resource from a parent's slot, if it
+	/// is equipped. Unequip can be successful if
+	/// - Item has been burned
+	/// - Item is equipped and extrinsic called by equipping item owner
+	/// - Item is equipped and extrinsic called by equipper NFT owner
+	/// Equipping operations are maintained inside the Equippings storage.
+	/// Modeled after [equip interaction](https://github.com/rmrk-team/rmrk-spec/blob/master/standards/rmrk2.0.0/interactions/equip.md)
+	///
+	/// Parameters:
+	/// - issuer: The caller of the function, not necessarily anything else
+	/// - item: Child NFT being equipped (or unequipped)
+	/// - equipper: Parent NFT which will equip (or unequip) the item
+	/// - base_id: ID of the equipped item's base
+	/// - slot_id: ID of the equipped item's slot
+	fn do_unequip(
+		issuer: T::AccountId,
+		item: (CollectionId, NftId),
+		equipper: (CollectionId, NftId),
+		base_id: BaseId,
+		slot_id: SlotId,
+	) -> Result<(CollectionId, NftId, BaseId, SlotId), DispatchError> {
+		let item_collection_id = item.0;
+		let item_nft_id = item.1;
+		let equipper_collection_id = equipper.0;
+		let equipper_nft_id = equipper.1;
+		// Check item NFT lock status
+		ensure!(
+			!pallet_rmrk_core::Pallet::<T>::is_locked(item_collection_id, item_nft_id),
+			pallet_uniques::Error::<T>::Locked
+		);
+		// Check equipper NFT lock status
+		ensure!(
+			!pallet_rmrk_core::Pallet::<T>::is_locked(equipper_collection_id, equipper_nft_id),
+			pallet_uniques::Error::<T>::Locked
+		);
+
+		let item_is_equipped =
```

Comment:
Since we repeat these 2 steps, should we create a helper functions like `is_nft_equipped((collection_id, nft_id), base_id, slot_id)` and `nft_exists(collection_id, nft_id)`? Could be useful for projects downstream in the future that use RMRK pallet as well. What do you think?

Reply:
i went ahead and did this.  initially i included returning as a Result, but that didn't work out, because...in the case of nft_exists, there was more action i needed to take; and in the case of `is_nft_equipped`, i couldn't throw an error because the two times we use it are in opposite cases, so the errors would be different (AlreadyEquipped in the case of `equip`, and NotEquipped in the case of `unequip`).  But anyway, the more helper methods the better.  Always appreciate the feedback and suggestions, as I'm just making it up as I go 👯

Reply:
done in [e138584](https://github.com/rmrk-team/rmrk-substrate/pull/190/commits/e138584b7bd13ab790f27fa41aa46072c08ef5ed)

Reply:
> i went ahead and did this. initially i included returning as a Result, but that didn't work out, because...in the case of nft_exists, there was more action i needed to take; and in the case of `is_nft_equipped`, i couldn't throw an error because the two times we use it are in opposite cases, so the errors would be different (AlreadyEquipped in the case of `equip`, and NotEquipped in the case of `unequip`). But anyway, the more helper methods the better. Always appreciate the feedback and suggestions, as I'm just making it up as I go dancers

I see, I think returning a `bool` is the right solution & this allows for others to return the relevant error based on their downstream implementation.